libcloudforensics/providers/aws/internal/ebs.py

```diff
@@ -305,3 +305,308 @@ def ShareWithAWSAccount(self, aws_account_id: str) -> None:
+      # ratio of 50 IOPS per 1 GiB.
+      create_volume_args['Iops'] = self.aws_account.ResourceApi(
+          common.EC2_SERVICE).Snapshot(snapshot.snapshot_id).volume_size * 50
+    try:
+      volume = client.create_volume(**create_volume_args)
+      volume_id = volume['VolumeId']
+      # Wait for volume creation completion
+      client.get_waiter('volume_available').wait(VolumeIds=[volume_id])
+    except (client.exceptions.ClientError,
+            botocore.exceptions.WaiterError) as exception:
+      raise RuntimeError('Could not create volume {0:s} from snapshot '
+                         '{1:s}: {2:s}'.format(volume_name, snapshot.name,
+                                               str(exception)))
+
+    zone = volume['AvailabilityZone']
+    encrypted = volume['Encrypted']
+
+    return AWSVolume(volume_id,
+                     self.aws_account,
+                     self.aws_account.default_region,
+                     zone,
+                     encrypted,
+                     name=volume_name)
+
+  def GetAccountInformation(self, info: str) -> str:
+    """Get information about the AWS account in use.
+
+    If the call succeeds, then the response from the STS API is expected to
+    have the following entries:
+      - UserId
+      - Account
+      - Arn
+    See https://boto3.amazonaws.com/v1/documentation/api/1.9.42/reference/services/sts.html#STS.Client.get_caller_identity for more details. # pylint: disable=line-too-long
+
+    Args:
+      info (str): The account information to retrieve. Must be one of [UserID,
+          Account, Arn]
+    Returns:
+      str: The information requested.
+
+    Raises:
+      KeyError: If the requested information doesn't exist.
+    """
+    account_information = self.aws_account.ClientApi(
+        common.ACCOUNT_SERVICE).get_caller_identity()  # type: Dict[str, str]
+    if not account_information.get(info):
+      raise KeyError('Key must be one of ["UserId", "Account", "Arn"]')
+    return account_information[info]
+
+  def _GenerateVolumeName(self,
+                          snapshot: AWSSnapshot,
+                          volume_name_prefix: Optional[str] = None) -> str:
+    """Generate a new volume name given a volume's snapshot.
+
+    Args:
+      snapshot (AWSSnapshot): A volume's Snapshot.
+      volume_name_prefix (str): Optional. Prefix for the volume name.
+
+    Returns:
+      str: A name for the volume.
+
+    Raises:
+      ValueError: If the volume name does not comply with the RegEx.
+    """
+
+    # Max length of tag values in AWS is 255 characters
+    user_id = self.GetAccountInformation('UserId')
+    volume_id = user_id + snapshot.volume.volume_id
+    volume_id_crc32 = '{0:08x}'.format(
+        binascii.crc32(volume_id.encode()) & 0xffffffff)
+    truncate_at = 255 - len(volume_id_crc32) - len('-copy') - 1
+    assert snapshot.name  # Mypy: assert that snapshot.name is not None
```

Comment:
I think these should be `is not None` globally.
Firstly it means the function could also raise an AssertionError but also Assertions are normally used for debugging and can be disabled globally based on runtime or environment variables

Reply:
Replaced by: 
```
if not snapshot.name:
  snapshot.name = snapshot.snapshot_id
```
In practice this method is never called with `snapshot.name = None`, this is only to satisfy `mypy`.

Reply:
I still don't think having assert statements in production code is a great idea.
Even if it's to satisfy the checker, it doesn't hurt us to have an actual if statement, that has no danger of being bypassed by someone's environment.

Reply:
@Fryyyyy as discussed earlier, replaced other occurrences. PTAL